From 75f0f4eb1e4f2823c39fe27137f78ac2c10bc293 Mon Sep 17 00:00:00 2001
From: Kirill Burtsev <kirill.burtsev@qt.io>
Date: Thu, 31 Mar 2022 19:45:39 +0200
Subject: [PATCH] [FIXUP] Fix compilation with system ICU

As of chromium 92.0.4480.0 after a set of fixes ending with a
https://chromium-review.googlesource.com/c/chromium/src/+/2830820
the need to override UCHAR_TYPE was dropped with the result that
codebase is ready for default 'char16_t'. Override was already dropped
for 98-based, which already builds fine for system ICU 70.1

Task-number: QTBUG-100495
Change-Id: I0e6d422f77b77015840647b52247f51736dd8a41
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
---

diff --git a/src/3rdparty/chromium/third_party/icu/BUILD.gn b/src/3rdparty/chromium/third_party/icu/BUILD.gn
index d5b1361..269de02 100644
--- a/src/3rdparty/chromium/third_party/icu/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/icu/BUILD.gn
@@ -410,12 +410,6 @@
     "USING_SYSTEM_ICU=1",
     "ICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC",
   ]
-
-  if (is_win) {
-    defines += [ "UCHAR_TYPE=wchar_t" ]
-  } else {
-    defines += [ "UCHAR_TYPE=uint16_t" ]
-  }
 }
 
 if (use_system_icu) {
